### PR TITLE
Migrate constants to Koanf

### PR DESCRIPTION
**Pull request by ccremer**
New internal configuration library [Koanf](https://github.com/knadh/koanf) (pronouned "konf") adds more flexibility into configuring the K8up operator itself.

* Introduces a new Configuration struct, that allows easier access than the tons of public Getters
* Koanf can be configured to also respect CLI flags (if there are any), config files etc with flexible merging hierarchy. It's not used yet, but it allows something similar to Viper, but it's more flexible and lightweight.

### CODE REVIEW

**Comment by ccremer** [comment]
@susana-garcia Do you have permissions to push to this branch?

**Comment by susana-garcia** [comment]
@ccremer no, I don't

**Comment by Kidswiss** [comment]
Makes sense to use some small framework to manage the configuration, although a ton of the current envVars will die with #121.

File: cfg/config.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+	PodFilter                      string `koanf:"podfilter"`
+	PromURL                        string `koanf:"promurl"`
+	RestartPolicy                  string `koanf:"restartpolicy"`
+}
+
+var (
+	Config = NewDefaultConfig()
+)
+
+const (
```
Comment (susana-garcia) [inline cfg/config.go:34]: I'd add the const block at the beginning

**Review by susana-garcia — COMMENTED** [review]
Looks good to me, I just add a small comment

**Pull request merged**